data/soapvalidator/Mail/Persona/SendOnBehalfOfMailUsingPrimaryAccount.xml

```diff
@@ -0,0 +1,372 @@
+					<right xmlns="">sendOnBehalfOf</right>
+				</GrantRightRequest>
+			</t:request>
+		<t:response>
+			<t:select path="//admin:GrantRightResponse" />
+		</t:response>
+	</t:test>	
+	</t:test_case>
+
+	<t:property name="server.zimbraAdmin" value="${zimbraServer.name}" />
+
+	<t:test_case testcaseid="sendMailOnBehalfOfPrimaryAccount" type="bhr">
+		<t:objective>Verify user is able to send an email with sendAs right
+		</t:objective>
+		<t:steps>
+			1. Login into Admin.
+			2. Grant Permission "sendAs" to Account1 on Account2
+			3. Login into Account1.
+			4. Send DiscoverRightsRequest to discover grants granted.
+			5. Send email with From: Primary Account to Account3 and Verify message details using GetMsgRequest
+			6. Verify message details using GetMsgRequest
+		</t:steps>
+
+		<!-- Login as gratee test_account1.name -->
+		<t:property name="server.zimbraAccount" value="${test_acct1.server}" />
+		<t:test id="auth_acct1" required="true" depends="acct1_setup">
+			<t:request>
+				<AuthRequest xmlns="urn:zimbraAccount">
+					<account by="name">${test_account1.name}</account>
+					<password>${defaultpassword.value}</password>
+				</AuthRequest>
+			</t:request>
+			<t:response>
+				<t:select path="//acct:AuthResponse/acct:lifetime" match="^\d+$" />
+				<t:select path="//acct:AuthResponse/acct:authToken" set="authToken" />
+			</t:response>
+		</t:test>
+
+		<!-- verify right using DiscoverRightsRequest -->
+		<t:test>
+			<t:request>
+				<DiscoverRightsRequest xmlns="urn:zimbraAccount">
+					<right>sendOnBehalfOf</right>
+				</DiscoverRightsRequest>
+			</t:request>
+			<t:response>
+				<t:select path="//acct:DiscoverRightsResponse/acct:targets"
+					attr="right" match="sendOnBehalfOf" />
+				<t:select
+					path="//acct:DiscoverRightsResponse/acct:targets/acct:target/acct:email[@addr='${test_account2.name}']" />
+			</t:response>
+		</t:test>
+
+		<!-- Send email with From: Primary Account -->
+		<t:test id="sendmail1" required="true">
+			<t:request>
+				<SendMsgRequest xmlns="urn:zimbraMail">
+					<m>
+						<e t="f" a="${test_account2.name}" />
+						<e t="t" a="${test_account3.name}" />
+						<su> ${msg01.subject}</su>
+						<mp ct="text/plain">
+							<content> ${message.content1}</content>
+						</mp>
+					</m>
+				</SendMsgRequest>
+			</t:request>
+			<t:response>
+				<t:select path="//zimbra:Code" match="service.PERM_DENIED"
+					emptyset="1" />
+				<t:select path="//mail:SendMsgResponse/mail:m" attr="id"
+					set="message.id1" />
+			</t:response>
+		</t:test>
+
+		<!-- Verify message details using GetMsgRequest -->
+		<t:test id="GetMsgRequest">
+			<t:request>
+				<GetMsgRequest xmlns="urn:zimbraMail">
+					<m id="${message.id1}" />
+				</GetMsgRequest>
+			</t:request>
+			<t:response>
+				<t:select path="//mail:GetMsgResponse/mail:m[@id='${message.id1}']">
+					<t:select path="//mail:e[@t='f']" attr="a"
+						match="${test_account2.name}" />
+					<t:select path="//mail:e[@t='t']" attr="a"
```

Comment:
Add validation for checking @t='s' element. In case of SOBO, sender value will be updated in this element.

Reply:
Added validation for element  @t='s'